### ISSUE TITLE
Add support for templates save for FSE Editor.

### ISSUE THREAD

**Issue by preda-bogdan** [Codeinwp/templates-patterns-collection#248]
### What problem does this address?

Currently, there is no support to save FSE templates from within the `site-editor` as the save feature relies on the postType being defined. Hence the template save is disabled as a result of #246

### What is your proposed solution?

We would need to improve support for the FSE editor and save FSE template data under a global option instead of the post meta, just for the FSE templates.
Also, we would need to check the import workflow and use another type `site-editor` instead of `gutenberg` to differentiate FSE templates from regular templates.

**Comment by HardeepAsrani** [comment]
Not a priority but will appreciate how you see this working whenever you have time to think about it @Codeinwp/design-team

**Comment by mghenciu** [comment]
This is a good idea. Not sure if Core plans to add a flow for this directly in FSE.
I know they are working on [Exporting/Importing Global Styles from other themes without changing it.](https://github.com/WordPress/gutenberg/issues/45371)

**Comment by mghenciu** [comment]
From a Design Perspective, what would be the requirements?

I guess one thing would be an additional Tab under My Library, for FSE Templates:
<img width="1274" alt="image" src="https://github.com/Codeinwp/templates-patterns-collection/assets/52494172/8dd13537-06e4-4074-ad07-33492474e6c3">

**Comment by HardeepAsrani** [comment]
@JohnPixle @mghenciu At this point, we can maybe consider renaming Gutenberg to Block Editor to avoid this confusion and the FSE theme being part of "FSE Theme" based on if the user is currently using an FSE theme.

**Comment by preda-bogdan** [comment]
@JohnPixle I would hide the FSE templates if the theme being used is not FSE. I like the idea of a toggle and using tags to differentiate them, and I agree that just having them under the Gutenberg tab should be enough as to not create more confusion. They are basically using the same editor but within a different context. 

For users that don't have FSE it will look the same, for the ones that have an FSE theme they will get some additional templates displayed.

**Comment by mghenciu** [comment]
sorry for the late replay. 
As Bogdan mentioned, FSE and Gutenberg templates are the same thing 99%. Maybe just the new elements like template parts are different but we don't use those.

And I wanted to mention that this issue refers to the **save for FSE Editor** in Templates Cloud. An action that it's usually taken by the user, so assume if they are using this feature of Save/Import -> then it's a FSE site.
So I think having the toggle should be enough, and just making sure we don't crash the site if they are not using FSE but import a FSE template (even if they do this un purpose).

@JohnPixle, what do you think if we reuse the same design from demosites.io, bu instead of PRO we write FSE with #325ce8?
![Screenshot 2023-05-16 at 13 15 08](https://github.com/Codeinwp/templates-patterns-collection/assets/52494172/ea458717-2bdf-435c-b0dd-63fd24399c86)

Based on how My Library pages look now (attached below), I think it will fit well: 
![image](https://github.com/Codeinwp/templates-patterns-collection/assets/52494172/9fde5710-ee47-4d50-95e1-08d660b3fb25)



Later we can see how and when we'll add FSE templates as Page Templates.

**Comment by mghenciu** [comment]
> And also, what will happen if a template is both FSE and PREMIUM? Is this a scenario?

Your are right John, we should think about this also.

**Essentially there would be 2 places where we show Templates:**

1. In **My Library TPC**, where users have the option to save their templates. Right now this covers Gutenberg, Elementor, Beaver and we want to add FSE now. This templates are bound to a license key, and can only be used under that license.
As a user, I would expect to see all my templates, even the ones that are not supported.

In this case, even now we show Beaver Builder & Elementor Tabs and templates - even if the users didn't installed the plugins/builders. But on import we notify the user that the builder will be installed and activated, in case it's not found (attached example from Beaver). 
<img width="1562" alt="image" src="https://github.com/Codeinwp/templates-patterns-collection/assets/52494172/a2f73695-90c7-458f-8620-dde0218df0ba">

**For FSE I don't think we can do something, if the user has a no FSE theme.**

___

2. On **Page Templates section** in TPC, available for all users. 
Here, we may add FSE templates later, but same UI should be respected - meaning that templates can be made in 3 different builders (excluding FSE) and at some point we may have Pro templates.

Based on this 2 contexts, I think what we can do for FSE (sincer there's no way to activate it), if a site has no FSE theme -> **we can show the templates - with a FSE label, but with a disabled state.** 

Another idea could be to show a notice like the one from above, but recommend one of our FSE themes.

Let me know if this makes sense.

**Comment by preda-bogdan** [comment]
@JohnPixle yes, it can be done, if you decide we want to show them with a disabled state for when an FSE theme is not active it should be possible.
